whisper-dictation.py

```diff
@@ -114,21 +126,35 @@ def __init__(self, recorder, languages=None, max_time=None):
         self.menu = menu
         self.menu['Stop Recording'].set_callback(None)
 
-        self.started = False
+        self.is_recording = False
+        self.is_transcribing = False
         self.recorder = recorder
         self.max_time = max_time
         self.timer = None
         self.elapsed_time = 0
+        self.recorder.set_transcription_events_listener(self)
```

Comment:
Looking a bit more through the code I think the way I structured it is a bit messy. There isn't really a good reason why the transcriber is passed into the recorder. I think the way it should be set up is:

```
class Recorder:
    def __init__(self):
        self.recording = False
  
    def start(self, callback, language=None):
        thread = threading.Thread(target=self._record_impl, args=(callback, language,))
        thread.start()

    def _record_impl(self, callback, language):
        ...
        callback(audio_data_fp32, language)



class StatusBarApp(..):

  def start_app(self, _): 
    def _callback(audio_data_fp32):
       self.is_transcribing = True
       self.update_title()
       self.transcriber.transcribe(audio_data_fp32, language)
       self.is_transcribing = False
       self.update_title()
      
    self.recorder.start(_callback, self.current_language)
    

if __name__ == "__main__":
    ...
    transcriber = SpeechTranscriber(model)
    recorder = Recorder()
    
    app = StatusBarApp(transcriber, recorder, args.language, args.max_time)
```

This would avoid all the listeners

Reply:
Yep, now code looks much better, good note)